### DISCUSSION TITLE
Getting ConnectionAbortedError after all nodes are finished subscription which was earlier working fine for last few months

### DISCUSSION THREAD

**Discussion by rajnishnationfirst** [FreeOpcUa/python-opcua#1456] [General]
Hi Team,

We developed a OPC UA client in python for datachange event for a SIEMENS WinCC over OPC UA  which was running fine for few 7-8 months and suddenly from today  where i am subscribing for all 900 nodes and then after the subscription and publiishing , ConnectionAbortedError showing the below exception is shown ..

Exceptions Thrown after all publishing is completed with Subscrption 

**ConnectionAbortedError: [WinError 10053] An established connection was aborted by the software in host machine.**

Image Snapshot of OPC UA Alarms Client

![skodaaaa](https://user-images.githubusercontent.com/64892368/164217171-6fd17d24-5456-4315-89db-34d3fa13d534.PNG)

Hi Team,

I am facing issues at start during subscribing 900+ Tags for on_datachnage_notiification event. ( here i have a python list containing all 900+ Node Ids of Tags and i am suscribing the whole list as shown below in the code .)

After subscription events  for all nodes are over then i am getting a ConnectionAbortedError which is shown in the image below .

if __name__ == "__main__":
    logging.basicConfig(level=logging.WARN)
    # logger = logging.getLogger("KeepAlive")
    # logger.setLevel(logging.DEBUG)

    client = Client("opc.tcp://AAAAAAAAAA:4861")

    try:
        client.connect()

        node_Id_List = []
        for iterator in range(len(tags_node_ids_list)):
            node_Id_List.append(client.get_node(tags_node_ids_list[iterator]))

   
        # subscribing to a variable node
        handler = SubHandler()
        sub = client.create_subscription(500, handler)
        handle = sub.subscribe_data_change(node_Id_List)
        time.sleep(0.1)

    finally:
        pass

Pls help  me colleagues.
Thanks 

Rajnish Vishwakarma

**Comment by AndreasHeine** [comment]
hey! i convert it to a discussion...

WinError 10053 is not an OPC UA Error!

**Reply by rajnishnationfirst** [reply]
Thanks Andreas for your reply ,

As you can see , the exception in the above image is due to user callback. How  can we solve this weird problem

**Reply by AndreasHeine** [reply]
not sure if it is possible to catch a error below socket level its not part of this library...
you can try to catch your whole app and restart if it happens... looks like network issue as mentioned!

**Reply by swamper123** [reply]
The error says, that a connection was aborted by the software on host computer side. 
The Error was risen by the OS, correct, but caused by sotware (and in worst case this unmaintained library).

Perhaps it was a future timeout, in the traceback is a cancelled future error, but what caused this cancelling is unclear. Maybe a timeout? Maybe sth. else. 
But I can't believe that after months of running and no changes made to it, that sth. broke in such a way (there are rare corner cases which may cause this, but pretty unlikely).

**Reply by rajnishnationfirst** [reply]
Thanks Swamper ... for your kind reply.

How can i increase timeouts and whihc timeouts ?

Where to change and increase the timeouts -- is it ua_client.py or client.py or server.py ?

Pls help coolleagues.. .. i am worrried on this and tried my level best from last 4 days but no breakthroughs.

Thanks

Rajnish Vishwakarma

**Reply by swamper123** [reply]
At this point, I, and I would guess others as well, can't help you.
We don't know your full code or changes you may have done.
We can't even say if this is a timeout or cancelled in another way. 
Was the future cancelling the cause of the WIN Error or the other way round?

There are many possibilities for your issue. And this library is not maintained anymore, even if sth. is wrong in the library background.
Either sb. else knows exactly what happens here and may answer you for free or you may need to pay sb. to diagnose your setup.
